### PR TITLE
refactor: account inputs

### PR DESCRIPTION
**Pull request by DominicGBauer**
## Description
Refactor account inputs to avoid future potential issues

## Work Done
* Updated from class to function component
* Changed a potential issue with + number and string

## How to Test
* Run the app
* Go to create account 
* Input an account (it should work as before)
* Go to send 
* Input an account to send to (it should work as before)

### CODE REVIEW

File: packages/frontend/src/components/accounts/AccountFormAccountId.js [hunk 1]
```diff
@@ -50,229 +51,208 @@ const InputWrapper = styled.div`
...
-
-        if (defaultAccountId) {
-            this.handleChangeAccountId({ userValue: accountId });
+const AccountFormAccountId = (props) => {
+    const [accountId, setAccountId] = useState(props.defaultAccountId || '');
+    const [invalidAccountIdLength, setInvalidAccountIdLength] = useState(false);
+    const [wrongChar, setWrongChar] = useState(false);
+    const debouncedAccountId = useDebouncedValue(accountId, ACCOUNT_CHECK_TIMEOUT);
+    const canvas = useRef(null);
+    const suffix = createRef();
```
Comment (lostpebble) [inline packages/frontend/src/components/accounts/AccountFormAccountId.js:61]: We need to make sure we are using the correct standards, especially when converting now from Class to Functional components. A lot of things don't convert one-to-one. Please double-check these things, as there are often nuances which can make a big difference. Almost all "utility" things with Functional components will start with `use...`.

(https://blog.logrocket.com/complete-guide-react-refs/#useref-vs-create-ref)

File: packages/frontend/src/components/accounts/AccountFormAccountId.js [hunk 2]
```diff
@@ -50,229 +51,208 @@ const InputWrapper = styled.div`
...
+    };
 
-        this.checkAccountAvailabilityTimer && clearTimeout(this.checkAccountAvailabilityTimer);
-        this.checkAccountAvailabilityTimer = setTimeout(() => {
-            this.handleCheckAvailability(accountId, type);
-        }, ACCOUNT_CHECK_TIMEOUT);
-    }
-
-    checkAccountIdLength = (accountId) => {
+    const checkAccountIdLength = (accountId) => {
```
Comment (lostpebble) [inline packages/frontend/src/components/accounts/AccountFormAccountId.js:124]: Don't like the naming of this function. For the boolean response, what question is this function asking? Its ambiguous here.

**Review by lostpebble — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/accounts/AccountFormAccountId.js (hunk 1 above)
Comment (lostpebble) [reply packages/frontend/src/components/accounts/AccountFormAccountId.js:61]: It helps to also look into the code sometimes. If you look into the code for `createRef()` you would see its not going to work out for a Functional Component paradigm.

File: packages/frontend/src/components/accounts/AccountFormAccountId.js (hunk 2 above)
Comment (DominicGBauer) [reply packages/frontend/src/components/accounts/AccountFormAccountId.js]: I just left the function names as they were previously, but have changed the name to be more descriptive.

File: packages/frontend/src/components/accounts/SetupSeedPhraseVerify.js [hunk 3]
```diff
@@ -72,66 +72,74 @@ const SetupSeedPhraseVerify = (
...
-                    data={{ wordId: wordId + 1 }}
-                />
+                <SafeTranslate id="input.enterWord.title" data={{ wordId: wordId + 1 }} />
             </h4>
             <Translate>
                 {({ translate }) => (
                     <input
                         data-test-id="seedPhraseVerificationWordInput"
-                        name='enterWord'
+                        name="enterWord"
```
Comment (lostpebble) [inline packages/frontend/src/components/accounts/SetupSeedPhraseVerify.js:93]: I've noticed a few double-quote symbols in this file. ESLint used to give us linting errors about that, but not it seems it doesn't care.

To be consistent with the old repo, we should keep the old rules and way they styled things. So please try get the linting working to ensure single-quotes are used for strings.

Prettier is still set correctly, so maybe we also just need to run that on a commit hook. But also don't want too many files to change all of a sudden now though- so the lint warning should suffice.

File: packages/frontend/src/components/accounts/AccountFormAccountId.tsx [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+                className={classNames([
+                    type,
+                    { success: success },
+                    { problem: problem },
+                    { 'wrong-char': wrongChar },
+                ])}
+            >
+                <input
+                    name='accountId'
+                    data-test-id="createAccount.accountIdInput"
```
Comment (lostpebble) [inline packages/frontend/src/components/accounts/AccountFormAccountId.tsx:219]: Another example mix of double-quotes with single-quotes

File: packages/frontend/src/components/common/LocalAlertBox.tsx [hunk 5]
```diff
@@ -56,19 +57,16 @@ const LocalAlertBoxContainer = styled.div`
...
 
-/**
- * Renders request status.
- *
- * @param localAlert {object} request status, can be null in case not completed yet / no outgoing request
- * @param localAlert.success {boolean} true if request was succesful
- * @param localAlert.messageCode {string} localization code of status message to display
- */
-const LocalAlertBox = ({ localAlert, accountId, dots }) => (
+interface Props {
```
Comment (lostpebble) [inline packages/frontend/src/components/common/LocalAlertBox.tsx:60]: I think we should maintain the documentation if possible (gives docs inline when using the component props). Generally a comment above each property should suffice, and one above the whole interface. If that doesn't work, then look into `jsdoc` and TypeScript interactions.

File: packages/frontend/src/components/SafeTranslate.tsx [hunk 6]
```diff
@@ -24,9 +24,15 @@ function processDataValue(value) {
 
     // Otherwise we're gonna leave it well enough alone
     return value;
+};
+
+interface Props {
```
Comment (lostpebble) [inline packages/frontend/src/components/SafeTranslate.tsx:29]: Generally prefer to place an upper case `I` character before interfaces to help differentiate them from other things. If it hasn't been done in this repo here though, don't worry about it.

**Review by lostpebble — CHANGES_REQUESTED** [review]

**Comment by edwardchew97** [comment]
Closing this PR due to lack of recent activity. Open to revisit this in the future.

**Pull request closed without merging**